### PR TITLE
HIVE-23683: Add queue time to compaction

### PR DESCRIPTION
**Pull request by pvary**
Added new QUEUE_TIME to the compaction tables.
Created upgrade scripts
Initialized the data with current timestamp
Made sure that the data is kept during the lifecycle
Updated HMS API to return the data
Created tests

### CODE REVIEW

File: ql/src/test/queries/clientpositive/sysdb.q [hunk 1]
```diff
@@ -1,5 +1,6 @@
 --! qt:dataset:src,part,srcbucket:ONLY
 --! qt:sysdb
+--! qt:replace:/(initiated\s+NULL\s+NULL\s+)[0-9]*(\s+NULL)/$1#Masked#$2/
```
Comment (klcopp) [inline ql/src/test/queries/clientpositive/sysdb.q:3]: Nit: maybe add a comment here about what's going on, to save someone time down the road

File: ql/src/test/queries/clientpositive/dbtxnmgr_showlocks.q [hunk 2]
```diff
@@ -1,3 +1,4 @@
+ --! qt:replace:/(initiated\s+---\s+---\s+)[0-9]*(\s+---)/$1#Masked#$2/
```
Comment (klcopp) [inline ql/src/test/queries/clientpositive/dbtxnmgr_showlocks.q:1]: Nit: maybe add a comment here about what's going on, to save someone time down the road

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java [hunk 3]
```diff
@@ -81,6 +81,8 @@ private void writeHeader(DataOutputStream os) throws IOException {
     os.write(Utilities.tabCode);
     os.writeBytes("Worker");
     os.write(Utilities.tabCode);
+    os.writeBytes("Queue Time");
```
Comment (klcopp) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java:84]: "Queue time" sounds like "time spent in queue." Something like "Entered queue" sounds more like a timestamp to me.

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java [hunk 4]
```diff
@@ -116,6 +117,7 @@ public String toString() {
       "partName:" + partName + "," +
       "state:" + state + "," +
       "type:" + type + "," +
+      "queueTime:" + queueTime + "," +
```
Comment (klcopp) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java:120]: (Opportunity: "start" is missing from this method)

File: ql/src/test/queries/clientpositive/sysdb.q (hunk 1 above)
Comment (pvary) [reply ql/src/test/queries/clientpositive/sysdb.q:4]: Done

File: ql/src/test/queries/clientpositive/dbtxnmgr_showlocks.q (hunk 2 above)
Comment (pvary) [reply ql/src/test/queries/clientpositive/dbtxnmgr_showlocks.q]: Done

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java (hunk 3 above)
Comment (pvary) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java]: How about enqueue time? Short, easy to understand...

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java (hunk 4 above)
Comment (pvary) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java]: Done

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java (hunk 3 above)
Comment (klcopp) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java]: Sounds good to me!

**Comment by pvary** [comment]
Thanks @klcopp and @laszlopinter86 for the reviews!

> One more: have you considered displaying a human-readable timestamp (maybe specifying UTC) in show compactions output? 

I decided against it, because every timestamp now is printed in epoch time in the show compactions output, and also I think this might break some user scripts if we change this behavior. Maybe in another jira with its' own config would be beneficial to have this.

> Also I recommend running the metastore upgrade tests if you haven't yet.

That was big help in identifying issues (happily not in this script) - See: https://issues.apache.org/jira/browse/HIVE-23697

**Pull request merged**